Yoast/wordpress-seo
# Use tabbed layout in the network settings screen

## Summary

This PR can be summarized in the following changelog entry:

* Implements the same tabbed layout in the plugin's network settings screen that is also used in the plugin's site settings screens.

## Relevant technical choices:

* This is the follow-up PR to #9690. It should be merged into the feature branch for #9447.
* A new `Yoast_Network_Admin` class is introduced that contains network-admin specific hooks and utility methods. It holds for example the method for getting a list of site choices (e.g. for dropdowns), and it also takes care of the restoring site functionality (both of which were previously "inline functions" in the `network.php` screen template file. This class is only initialized if the current context is a multisite and we're currently in the network admin.
* A few improvements over the original introduction of the network settings API in #9690 needed to be made: The network settings API is not only scoped to the network admin screens, it might also be used in other areas, for example the REST API or WP-CLI. Therefore the network admin-specific parts of `Yoast_Network_Settings_API` were relocated to the new `Yoast_Network_Admin` class. This is in line with how core now allows registering settings from anywhere in the codebase, while the admin parts of the settings API are only available in the admin context. Two simple getter methods have been added to `Yoast_Network_Settings_API` to have access to the registered and whitelisted settings from the outside.
* The `WPSEO_Option_Tabs` class is now used to power the network settings screen, and a single "General" tab has been added for now, holding the existing settings. As a result, there's now a separate template file for that tab. This now opens up flexibility, and more tabs will be added in the future as part of other enhancement issues.
* The page titles in the network admin have been adjusted to follow the same format as the page titles in the site admin ("Page Title - Yoast SEO" instead of "Yoast SEO: Page Title"). In addition, the term "MultiSite" has been replaced with "Network", similar how other core pages do use the latter term in favor of "multisite".
* The network admin pages have been adjusted to use a new `wpseo_manage_network_options` capability as requirement, which allows for more granular access management than with reusing an arbitrary core capability (previously `delete_users` was used). This is fully backward-compatible since a network administrator has all capabilities by default anyway.

## Test instructions

This PR can be tested by following these steps:

* Set up a multisite and activate the plugin network-wide.
* Go into the network admin.
* Ensure that you can access the plugin's three network admin pages.
* Ensure that the "General" settings page in the network admin uses a tabbed layout and looks in line with the site admin screens' appearance.
* Ensure that when you change the two available network settings, the values you set persist correctly.
* Ensure that the "Restore site to default settings" form still works correctly:
    * Go to the site admin and change some of the plugin's settings.
    * Go back to the network admin, select the site where you changed the settings, and submit the form.
    * After submitting, you should receive a success feedback message at the top of the screen.
    * Go back to the site admin and verify that the settings you changed are back to their defaults.

## Quality assurance


Fixes #9688
